components/family-studies/api/src/main/java/org/phenotips/studies/family/internal/AbstractBasePedigree.java

```diff
@@ -0,0 +1,114 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see http://www.gnu.org/licenses/
+ */
+package org.phenotips.studies.family.internal;
+
+import org.phenotips.studies.family.Pedigree;
+
+import org.apache.commons.lang3.StringUtils;
+import org.apache.commons.lang3.tuple.Pair;
+import org.json.JSONObject;
+
+/**
+ * A common base for classes supporting different pedigree formats.
+ *
+ * Common functionality (mostly related to images, which have the same format) is placed in this class.
+ *
+ * @version $Id$
+ */
+public abstract class AbstractBasePedigree implements Pedigree
+{
+    protected JSONObject data;
+
+    protected String image = "";
+
+    /**
+     * Create a new pedigree using given data and image.
+     *
+     * @param data pedigree data
+     * @param image SVG 'image'
+     */
+    public AbstractBasePedigree(JSONObject data, String image)
+    {
+        if (data == null || data.length() == 0) {
+            throw new IllegalArgumentException();
+        }
+        this.data = data;
+        this.image = image;
```

Comment:
not sure, probably not - when pedigree will be imported via REST there will be pedigree data but no image (yet)

..at that point we will have to make sure nothing crashes in that case downstream, but having no image is fine in itself